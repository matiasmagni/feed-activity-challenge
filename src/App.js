import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from "react-router-dom";
import Home from './features/Home/Home';
import Comments from './features/Comments/Comments';
import './App.css';

const App = () => {
  return (
    <Router>
      <header className="mt-0 pt-0">
        <div className="bg-cover clearfix pt-3">
          <h2 className="logo">Shahala</h2>
          <input type="text" id="nav-search" className="form-control nav-search mx-auto" name="" />
          <div className="ml-0 mr-0 pb-1">
            <nav className="navbar navbar-expand-md">
              <button className="navbar-toggler ml-auto" data-target="#my-nav" data-toggle="collapse"
                aria-controls="my-nav" aria-expanded="false"
                aria-label="Toggle navigation">
                <span className="bar1"></span> <span className="bar2"></span> <span className="bar3"></span>
              </button>
              <div id="my-nav" className="collapse navbar-collapse">
                <ul className="navbar-nav mx-auto">
                  <li className="nav-item">
                    <Link className="nav-link" to="/">Home</Link>
                  </li>
                  <li className="nav-item">
                    <form action="" method="POST">
                      <div className="input-group mt-0 mx-auto" style={{ width: 16 }}>
                        <div className="">
                          <img src="/images/search-icon.png" id="toggle-search"
                            className="ml-2 toggle-search" alt="search icon" />
                        </div>
                      </div>
                    </form>
                  </li>
                </ul>
              </div>
            </nav>
          </div>
        </div>
      </header>
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/comments/:postId">
          <Comments />
        </Route>
      </Switch>
    </Router>
  );
}

export default App;
