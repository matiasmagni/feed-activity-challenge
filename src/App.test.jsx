import React from 'react';
import { cleanup } from '@testing-library/react';
import { act as domAct } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { act as testAct, create } from 'react-test-renderer';
import store from './redux/store';
import App from './App';

afterEach(cleanup);

describe('App component tests.', () => {
  it('App component renders correctly.', () => {
    let tree;

    domAct(() => {
      testAct(() => {
        tree = create(<Provider store={store}><App /></Provider>);
      });

      expect(tree).toMatchSnapshot();
    });
  });
});
