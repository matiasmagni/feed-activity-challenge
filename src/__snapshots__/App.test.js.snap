// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests. App component renders correctly. 1`] = `
Array [
  <header
    className="mt-0 pt-0"
  >
    <div
      className="bg-cover clearfix pt-3"
    >
      <h2
        className="logo"
      >
        Shahala
      </h2>
      <input
        className="form-control nav-search mx-auto"
        id="nav-search"
        name=""
        type="text"
      />
      <div
        className="ml-0 mr-0 pb-1"
      >
        <nav
          className="navbar navbar-expand-md"
        >
          <button
            aria-controls="my-nav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler ml-auto"
            data-target="#my-nav"
            data-toggle="collapse"
          >
            <span
              className="bar1"
            />
             
            <span
              className="bar2"
            />
             
            <span
              className="bar3"
            />
          </button>
          <div
            className="collapse navbar-collapse"
            id="my-nav"
          >
            <ul
              className="navbar-nav mx-auto"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link"
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </li>
              <li
                className="nav-item"
              >
                <form
                  action=""
                  method="POST"
                >
                  <div
                    className="input-group mt-0 mx-auto"
                    style={
                      Object {
                        "width": 16,
                      }
                    }
                  >
                    <div
                      className=""
                    >
                      <img
                        alt="search icon"
                        className="ml-2 toggle-search"
                        id="toggle-search"
                        src="/images/search-icon.png"
                      />
                    </div>
                  </div>
                </form>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </div>
  </header>,
  <div
    className="container-fluid video-player"
  >
    <div
      className="container"
    >
      <div
        className="row vr-gallery"
      />
    </div>
  </div>,
  <footer
    className="container-fluid pt-5"
  >
    <div
      className="container"
    >
      <h2
        className="logo text-center"
      >
        Shahala
      </h2>
      <nav
        className="nav nav-fill mx-auto mt-5"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://facebook.com/fh5co"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-facebook-f"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://facebook.com/fh5co"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-twitter"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/#"
          >
            <i
              className="fab fa-instagram"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/#"
          >
            <i
              className="fab fa-google-plus-g"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/#"
          >
            <i
              className="fas fa-rss"
            />
          </a>
        </li>
      </nav>
    </div>
    <div
      className="copyright mt-4"
    >
      <p
        className="text-center"
      >
        Â© 2019 
        <a
          className="text-white"
          href="/#"
        >
          Shahala
        </a>
        . All Rights Reserved. Design by 
        <a
          className="text-white"
          href="https://freehtml5.co/shahala"
          rel="noreferrer"
          target="_blank"
        >
          FreeHTML5.co
        </a>
        .
      </p>
    </div>
  </footer>,
]
`;
